pandas/tests/groupby/test_apply.py

```diff
@@ -880,3 +880,24 @@ def test_apply_function_index_return(function):
         index=pd.Index([1, 2, 3], name="id"),
     )
     tm.assert_series_equal(result, expected)
+
+
```

Comment:
can you put this in a *new* file, call it ``test_apply_mutate.py``; there are 2 tests 

```
pandas/tests/groupby/test_groupby.py:def test_mutate_groups():
pandas/tests/groupby/test_groupby.py:def test_no_mutate_but_looks_like():
```

that should also be moved there.